### PR TITLE
[INFRA-515] Add new command to accept HOSTING requests

### PR DESCRIPTION
**Pull request by slide**
This will streamline the ability to accept HOSTING requests. I have not had a chance to test this because I don't have access to the actual bot that has the actual credentials. 

@batmat Please review this for sure

### CODE REVIEW

File: src/main/java/org/jenkinsci/backend/ircbot/IrcBotImpl.java [hunk 1]
```diff
@@ -202,6 +204,16 @@ private void handleDirectCommand(String channel, String sender, String login, St
             return;
         }
 
+        m = Pattern.compile("(?:host) (\\d+)((?:[ ]+)(\\S+))?", CASE_INSENSITIVE).matcher(payload);
+        if (m.matches()) {
+            String forkTo = "";
+            if(m.groupCount() > 1) {
+                forkTo = m.group(2);
+            }
+            setupHosting(channel,sender,m.group(1), forkTo);
```
Comment (batmat) [inline src/main/java/org/jenkinsci/backend/ircbot/IrcBotImpl.java]: NIT: irregular spaces after commas.

File: src/main/java/org/jenkinsci/backend/ircbot/IrcBotImpl.java [hunk 2]
```diff
@@ -260,6 +272,105 @@ private void kickUser(String channel, String sender, String target) {
...
+
+            String forkFrom = "";
+            List<String> users = new ArrayList<String>();
+
+            String defaultAssignee = issue.getReporter();
+
+            RemoteCustomFieldValue[] customFields = issue.getCustomFieldValues();
+            for(RemoteCustomFieldValue val : customFields) {
+                String fieldId = val.getCustomfieldId();
+                if(fieldId.equalsIgnoreCase("customfield_10320")) {
```
Comment (batmat) [inline src/main/java/org/jenkinsci/backend/ircbot/IrcBotImpl.java]: Create constants to clarify what those fields match?

**Comment by batmat** [comment]
Awesome idea!

I cannot test it, so it seems fine just reading it.

@rtyler @daniel-beck is there a way we could set up some access or something to test that? Thanks

**Comment by rtyler** [comment]
@batmat I don't remember the specifics of the environment variable overrides, but you can run the Docker container directly to test that it at least somewhat works.

Testing the functionality for JIRA operations specifically would require you to give the container your own credentials though.

**Comment by slide** [comment]
I'll try running this in the docker container and see if it works on a "bogus" hosting request.

**Comment by slide** [comment]
I tried running in docker and was able to run, but I could only really test the commenting on the issue. It did show me that some sort of error handling would be good so that the issue is not updated if something fails.

**Comment by oleg-nenashev** [comment]
I still have test environment for the bot, so I can test the PR on the next weekend

**Comment by slide** [comment]
@oleg-nenashev That would be great, thanks for the help!

**Comment by slide** [comment]
This most recent commit adds some error checking to the various steps so that it bails early if needed.

**Comment by slide** [comment]
The bot code may need to be updated in general based on https://issues.jenkins-ci.org/browse/INFRA-787

**Comment by oleg-nenashev** [comment]
@slide Yep, updating it

**Comment by oleg-nenashev** [comment]
Found the long-standing issue for it https://issues.jenkins-ci.org/browse/INFRA-515

**Pull request merged**